src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -1718,5 +1718,96 @@ private static bool TryFindFirstMatchedLane(Vector128<ushort> compareResult, ref
+                    Unsafe.As<byte, Vector256<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector256<byte>>(ref last) = tempFirst;
+
+                    // Adjust the references to point to the next vector
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector256<byte>.Count);
+                    numBytesWritten += Vector256<byte>.Count * 2;
+                } while (numBytes - numBytesWritten >= Vector256<byte>.Count * 2);
+            }
+            else if (Ssse3.IsSupported && Vector128<byte>.Count * 2 <= numBytes)
+            {
+                Vector128<byte> ReverseMask = Vector128.Create((byte)14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1);
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector128<byte>.Count);
+                do
+                {
+                    // Load in values from beginning and end of the array.
+                    Vector128<byte> tempFirst = Unsafe.As<byte, Vector128<byte>>(ref first);
+                    Vector128<byte> tempLast = Unsafe.As<byte, Vector128<byte>>(ref last);
+
+                    // Shuffle to reverse each vector:
+                    //     +-------------------------------+
+                    //     | A | B | C | D | E | F | G | H |
+                    //     +-------------------------------+
+                    //          --->
+                    //     +-------------------------------+
+                    //     | H | G | F | E | D | C | B | A |
+                    //     +-------------------------------+
+                    tempFirst = Ssse3.Shuffle(tempFirst, ReverseMask);
+                    tempLast = Ssse3.Shuffle(tempLast, ReverseMask);
+
+                    // Store the reversed vectors
+                    Unsafe.As<byte, Vector128<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector128<byte>>(ref last) = tempFirst;
+
+                    // Adjust the references to point to the next vector
+                    first = ref Unsafe.Add(ref first, Vector128<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector128<byte>.Count);
+                    numBytesWritten += Vector128<byte>.Count * 2;
+                } while ((numBytes - numBytesWritten) >= Vector128<byte>.Count * 2);
+            }
+
+            // Store any remaining values one-by-one
+            last = ref Unsafe.Add(ref first, (numBytes - numBytesWritten) - sizeof(char));
+            ref char firstChar = ref Unsafe.As<byte, char>(ref first);
+            ref char lastChar = ref Unsafe.As<byte, char>(ref last);
+            while (numBytes - numBytesWritten > sizeof(char))
+            {
+                (lastChar, firstChar) = (firstChar, lastChar);
+                firstChar = ref Unsafe.Add(ref firstChar, 1);
+                lastChar = ref Unsafe.Add(ref lastChar, -1);
```

Comment:
Semantically, using [Subtract](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.subtract?view=net-6.0) would be better.
```suggestion
                lastChar = ref Unsafe.Subtract(ref lastChar, 1);
```